catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java

```diff
@@ -104,6 +111,30 @@ public void destroy() {
         executorService.shutdown();
     }
 
+    private void updateIdentityNodeName(Metacard metacard) throws FederationAdminException {
+        RegistryPackageType registryPackage = federationAdminService.getRegistryObjectByRegistryId(
+                RegistryUtility.getRegistryId(metacard));
+
+        String siteName = SystemInfo.getSiteName();
+        RegistryPackageTypeHelper registryPackageTypeHelper = new RegistryPackageTypeHelper();
+
+        registryPackageTypeHelper.getExtrinsicObjects(registryPackage.getRegistryObjectList())
+                .stream()
+                .filter(extrinsicObjectType -> extrinsicObjectType.getObjectType()
+                        .equals(RegistryConstants.REGISTRY_NODE_OBJECT_TYPE))
+                .forEach(extrinsicObjectType -> extrinsicObjectType.setName(
```

Comment:
You can swap out `forEach` with `first` since there should only be one matching extrinsic object.